cmd/signing_ethereum.go

```diff
@@ -0,0 +1,225 @@
+					Name: "eth-key-file,k",
+					Usage: "Path to the ethereum key file. " +
+						"If not provided read the path from a config file.",
+				},
+				cli.StringFlag{
+					Name:  "output-file,o",
+					Usage: "Output file for the signature",
+				},
+			},
+		},
+		{
+			Name:        "verify",
+			Usage:       "Verifies a signature",
+			Description: ethereumVerifyDescription,
+			Action:      EthereumVerify,
+			ArgsUsage:   "[ethereum-signature]",
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "input-file,i",
+					Usage: "Input file with the signature",
+				},
+			},
+		},
+	},
+}
+
+const ethereumSignDescription = `Calculates an ethereum signature for a given message.
+The message is expected to be provided as a string, it is later hashed with SHA-256
+and passed to Ethereum ECDSA signing. Signature is calculated in Ethereum specific
+format as a hexadecimal string representation of 65-byte {R, S, V} parameters.
+
+It requires an Ethereum key to be provided in an encrypted file. A path to the key file
+can be configured in a config file or specified directly with an 'eth-key-file' flag.
+
+The key file is expected to be encrypted with a password provided as ` + config.PasswordEnvVariable + `
+environment variable.
+	
+The result is outputted in a common Ethereum signature format:
+{
+	"address": "<address>",
+	"msg": "<content>",
+	"sig": "<signature>",
+	"version": "2"
+}
+
+If 'output-file' flag is set the result will be stored in a specified file path.
+`
+
+const ethereumVerifyDescription = `Verifies if a signature was calculated for a message 
+by an ethereum account identified by an address. 
+
+It expects a signature to be provided in a common Ethereum signature format:
+{
+	"address": "<address>",
+	"msg": "<content>",
+	"sig": "<signature>",
+	"version": "2"
+}
+
+If 'input-file' flag is set the input will be read from a specified file path.
+`
+
+// EthereumSignature is a common Ethereum signature format.
+type EthereumSignature struct {
+	Address   common.Address `json:"address"`
+	Message   string         `json:"msg"`
+	Signature string         `json:"sig"`
+	Version   uint           `json:"version"`
+}
+
+const ethSignatureVersion uint = 2
+
+// EthereumSign signs a string using operator's ethereum key.
+func EthereumSign(c *cli.Context) error {
+	message := c.Args().First()
+	if len(message) == 0 {
+		return fmt.Errorf("invalid digest")
+	}
+
+	var ethKeyFilePath, ethKeyPassword string
+	// Check if `eth-key-file` flag was set. If not read the key file path from
+	// a config file.
+	if ethKeyFilePath = c.String("eth-key-file"); len(ethKeyFilePath) > 0 {
+		ethKeyPassword = os.Getenv(config.PasswordEnvVariable)
+	} else {
+		ethereumConfig, err := config.ReadEthereumConfig(c.GlobalString("config"))
+		if err != nil {
+			return fmt.Errorf("failed while reading config file: [%v]", err)
+		}
+
+		ethKeyFilePath = ethereumConfig.Account.KeyFile
+		ethKeyPassword = ethereumConfig.Account.KeyFilePassword
```

Comment:
Just to double check.. Do we allow passwords to be stored in config file? in `config.toml.SAMPLE` there is no password for eth account

Reply:
No, we don't. We read it from env variable and store in config struct https://github.com/keep-network/keep-ecdsa/blob/3fb44f02715abe806d41eae6fa4417994f8665b3/internal/config/config.go#L94

Reply:
Ok, I missed that part.